src/net/dash/segment_loader.ts

```diff
@@ -97,23 +92,15 @@ const segmentPreLoader = (customSegmentLoader? : CustomSegmentLoader) => ({
   period,
   representation,
   segment,
-} : ISegmentLoaderArguments) : ILoaderObservable<Uint8Array|ArrayBuffer> => {
-  const {
-    media,
-    range,
-    indexRange,
-    isInit,
-  } = segment;
-
-  // init segment without initialization media/range/indexRange:
-  // we do nothing on the network
-  if (isInit && !(media || range || indexRange)) {
-    return Observable.empty();
-  }
+} : ISegmentLoaderArguments) : ILoaderObservable<Uint8Array|ArrayBuffer|null> => {
+  const { mediaURL } = segment;
 
-  // construct url for the segment
-  const path = media ? replaceTokens(media, segment, representation) : "";
-  const url = resolveURL(representation.baseURL, path);
+  if (mediaURL == null) {
+    return observableOf({
+      type: "data" as "data",
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Remove this unnecessary cast. ([typescript:S4325](https://sonarcloud.io/organizations/rx-player/rules#rule_key=typescript%3AS4325))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=rx-player&pullRequest=254&issues=AWRfEDce1IJZDfsS9yTq&open=AWRfEDce1IJZDfsS9yTq)

[ISSUE_KEY]: AWRfEDce1IJZDfsS9yTq